src/root.tsx

```diff
@@ -12,12 +12,22 @@ import {
     Scripts,
     Title
 } from "solid-start";
+import i18next from "i18next";
+import i18n from './i18n/config';
 import "./root.css";
 import { Provider as MegaStoreProvider } from "~/state/megaStore";
 import { Toaster } from "~/components/Toaster";
 import ErrorDisplay from "./components/ErrorDisplay";
+import { I18nProvider } from "./components/I18nProvider";
 
 export default function Root() {
+    const [loaded, setLoaded] = createSignal(false);
+
+    onMount(async () => {
```

Comment:
I think you can clean this up a bit by putting the loading logic inside the provider. here's what I came up with:

```tsx
import { ParentComponent, Show, createResource } from "solid-js";
import { I18nContext } from "../i18n/context";
import i18next from "i18next";
import i18nConfig from "../i18n/config";

export const I18nProvider: ParentComponent = (props) => {
    const [i18nConfigured] = createResource(async () => {
        await i18nConfig;
        return true;
    });

    return (
        <Show when={i18nConfigured()}>
            <I18nContext.Provider value={i18next}>
                {props.children}
            </I18nContext.Provider>
        </Show>
    );
};
```

seems to get the same behavior and can just use

`<I18nProvider>` inside root without any props

(unless I'm missing some reason this def needs to be in root?)

Reply:
Oh, interesting solution. I implemented as you suggested and it worked like a charm. I am gonna mark this conversation as resolved as well.